Add configuration to camunda-platform7 and update metadata of Camunda related projects

tobiasschaefer opened the pull request:
I'm the lead developer of the Micronaut Camunda Integration Project.

Please merge this PR which adds configuration options and makes the setup easier.

I tested the following:
./gradlew micronaut-cli:run --args="create-app --list-features" -> Looks good, feature is listed

Both the following look good
./gradlew micronaut-cli:run --args="create-app --features=camunda-platform7 temp-gradle"
./gradlew micronaut-cli:run --args="create-app --build=maven --features=camunda-platform7 temp-maven"

and build successfully.

FYI: @sdelamo

tobiasschaefer committed:
Add configuration to camunda-platform7 and update metadata of Camunda related projects

sdelamo commented on starter-core/src/main/java/io/micronaut/starter/feature/camunda/Platform7.java:
```diff
@@ -23,17 +23,21 @@
+    private final AssertJ assertJ;
 
-    public Platform7(DatabaseDriverFeature defaultDbFeature, AssertJ assertJ) {
+    public Platform7(DatabaseDriverFeature defaultDbFeature, Jetty jetty, AssertJ assertJ) {
```
Annotate the new constructor with @Inject. Keep the old constructor deprecated.

tobiasschaefer replied:
I don't see why we need to keep the old constructor. IMHO it would never be called. We need the new constructor to initialize all fields.

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/feature/camunda/Platform7Spec.groovy:
```diff
@@ -83,8 +85,28 @@ class Platform7Spec extends ApplicationContextSpec implements CommandOutputFixtu
       <scope>runtime</scope>
     </dependency>
+''') == 1
+        template.count('''\
```
Please, use the new BuildVerifierApi to test dependencies.

tobiasschaefer replied:
Thanks for the hint.
I switched to BuildTestVerifier (I guess that is what you are referring to)

sdelamo commented on starter-core/src/main/java/io/micronaut/starter/feature/camunda/Platform7.java:
```diff
@@ -66,16 +70,26 @@ public boolean supports(ApplicationType applicationType) {
 
     @Override
     public void apply(GeneratorContext generatorContext) {
+        generatorContext.getConfiguration().put("camunda.admin-user.id", "admin");
```
extract `protected void addConfiguration(GeneratorContext)` method with all these statements.

sdelamo commented on starter-core/src/main/java/io/micronaut/starter/feature/camunda/Platform7.java:
```diff
@@ -66,16 +70,26 @@ public boolean supports(ApplicationType applicationType) {
         if (!featureContext.isPresent(DatabaseDriverFeature.class)) {
             featureContext.addFeature(defaultDbFeature);
         }
+        if (!featureContext.isPresent(Jetty.class)) {
```
use `FeatureContext::addFeatureIfNotPresent`

tobiasschaefer commented:
Hi @sdelamo ,

thank you for your comments in the PR.

I updated the PR to consider your comments.

Note: I updated `GradleBuildTestVerifier` to match the behaviour of `MavenBuildTestVerifier`, i.e. it ignores the dependency version information if the dependency is not managed. This is needed because all Camunda related dependencies are currently not managed by the micronaut-bom. If you give me hints how to accomplish this I'm happy to do it in a separate PR.

wetted commented:
> Note: I updated `GradleBuildTestVerifier` to match the behaviour of `MavenBuildTestVerifier`, i.e. it ignores the dependency version information if the dependency is not managed. This is needed because all Camunda related dependencies are currently not managed by the micronaut-bom. If you give me hints how to accomplish this I'm happy to do it in a separate PR.

@tobiasschaefer @sdelamo 
This change breaks `GradleBuildTestVerifier` and needs to be fixed. For example, if I have the following,

```
        verifier.hasDependency("io.micronaut.oraclecloud", "micronaut-oraclecloud-function-http", Scope.COMPILE)
        !verifier.hasDependency("io.micronaut.oraclecloud", "micronaut-oraclecloud-function")
```

The second check fails because the check for "micronaut-oraclecloud-function" is matching "micronaut-oraclecloud-function-http". The regex is flawed in that regard. I'll work on a fix for it in my PR. The motivation is just to ignore version notation in the dependency, right?

tobiasschaefer commented:
Hi @wetted ,

yes, the motivation is just to ignore version notation in the dependency (so that the Gradle Implementation matches the Maven Implementation).

Thanks for fixing the issue for the oracle-function-http feature 🥇 Strange though, that all tests were green.

wetted commented:
@tobiasschaefer the tests were green for you. I added a new test that exposed the need for a fix. :)
